### DISCUSSION TITLE
Tokens in instances

### DISCUSSION THREAD

**Discussion by LeJanusz** [six7/figma-tokens#445] [Q&A]
Quick question:
When i apply tokens within one file, everything works as expected. But when i try to import an instance in another file, all the tokens are gone and therefore nothing is changing.
Is this a limitation? 

When i just copy and paste the instance from the library file, it works. So, inserting components wont still work, which is a no-go for our use cases.

Components are published and updated.

**Comment by six7** [comment]
Hey @LeJanusz - I assume you're using `Local` as your sync method, which will store tokens on the current file. If you want to keep using those tokens outside of that file, you need to use a Sync method. 

Currently those involve third party services, such as JSONBin, GitHub or storing on your own server.

https://docs.tokens.studio/sync/sync

**Reply by LeJanusz** [reply]
I am using jsonbin. And all my tokens appear in the new file after I pulled them from JSONbin. But it seems the plugin is not seeing any reference to any token anymore. In the Master component there are "Figmatokens" listed in the right prop panel, but not when inserting an instance of the same component in a new file.

**Reply by six7** [reply]
Ah, got it. You mean that text at the bottom right? Those don't work reliably for instances as Figma doesn't allow us to set something there for instances. Tokens are still applied, though.
